### ISSUE TITLE
Profiles are reset after deactivated

### ISSUE THREAD

**Issue by nerun** [costales/gufw#61]
_Related to issues #25 (closed) and #54 (open), also PR #55 (closed without merge)._

* gufw 22.04.0-0ubuntu1

I have the same problem. All my rules were added by GUFW, they all have custom names. But as soon as I disable GUFW and reactivate it, all the rules become immutable, as if they were created by UFW and not by GUFW.

I think this is a bug, not an issue in my system or corrupted files.

I have tested here:
1. I made a backup of my gufw profile: `sudo cp /etc/gufw/Início.profile /etc/gufw/Início.profile.bak`
2. Disabled and reactivated gufw: all rules become immutable as if they were created by UFW.
3. Disabled gufw again.
4. Restored profile backup: `sudo cp /etc/gufw/Início.profile.bak /etc/gufw/Início.profile`
5. Enabled gufw again: now all the rules are gufw and i can edit them again.

Look at this screenshot. Before disable GUFW and after reactivate it. The sizes changes. Profile is reseted!

![Captura de tela de 2023-07-10 10-57-41](https://github.com/costales/gufw/assets/3987342/5d99b20a-f0cb-46e2-961a-f5bc6ca37808)

I have tested to create a custom profile: `Daniel.profile`. But the problem allways happens.

### Also, PR #55 fixed this issue, but was closed without merge and any discussion.

**Comment by nerun** [comment]
### Also, PR #55 fixed this issue, but was closed without merge.

**Comment by nerun** [comment]
Uh, permissions to /etc/gufw are different. Files are 600 not 644.
```sh
$ exa -l --tree -L1 --octal-permissions --group-directories-first --no-time /etc/gufw
0755 drwxr-xr-x    - root /etc/gufw
0755 drwxr-xr-x    - root ├── app_profiles
0600 .rw------- 1,4k root ├── Daniel.profile
0600 .rw------- 1,4k root ├── Daniel.profile.bak
0600 .rw-------   76 root ├── Escritório.profile
0600 .rw-------   94 root ├── gufw.cfg
0600 .rw-------   80 root ├── Início.profile
0600 .rw-------   78 root └── Público.profile
```

**Comment by nerun** [comment]
I have restored the original `/usr/share/gufw/gufw/gufw/model/firewall.py` (PR #55 fix) to test if the issue was caused by wrong permissions NOT in `/usr/share/gufw` BUT in `/etc/gufw`. And...

Set ONLY `*.profiles` from octal permission 600 to:
* 640 = no change
* 644 = fixed the issue !!!

**Comment by nerun** [comment]
Change profile permissions in `/etc/gufw` fix the issue just once. When it is disabled again files are deleted, reset and permissions get back to 600.

**Comment by costales** [comment]
Hi, thank for your feedback! Is this PR fixing it? https://github.com/costales/gufw/pull/55

**Comment by nerun** [comment]
> Hi, thank for your feedback! Is this PR fixing it? #55

Yes, it is. I am using this in my system.

**Comment by costales** [comment]
This will be fixed in next release (24.04). Thank you for your feedback!!
